discoverx/delta_housekeeping.py

```diff
@@ -0,0 +1,368 @@
+            ))
+        )
+        # max & 2nd timestamp of OPTIMIZE into output
+        out = describe_detail_df.join(
+            operation_order
+            .filter((F.col("operation") == "OPTIMIZE") & (F.col("operation_order") == 1))
+            .select("catalog", "database", "tableName", "timestamp")
+            .withColumnRenamed("timestamp", "max_optimize_timestamp"),
+            how="outer", on=["catalog", "database", "tableName"]
+        )
+        out = out.join(
+            operation_order
+            .filter((F.col("operation") == "OPTIMIZE") & (F.col("operation_order") == 2))
+            .select("catalog", "database", "tableName", "timestamp")
+            .withColumnRenamed("timestamp", "2nd_optimize_timestamp"),
+            how="outer", on=["catalog", "database", "tableName"]
+        )
+        # max timestamp of VACUUM into output
+        out = out.join(
+            operation_order
+            .filter((F.col("operation") == "VACUUM END") & (F.col("operation_order") == 1))
+            .select("catalog", "database", "tableName", "timestamp")
+            .withColumnRenamed("timestamp", "max_vacuum_timestamp"),
+            how="outer", on=["catalog", "database", "tableName"]
+        )
+        out = out.join(
+            operation_order
+            .filter((F.col("operation") == "VACUUM END") & (F.col("operation_order") == 2))
+            .select("catalog", "database", "tableName", "timestamp")
+            .withColumnRenamed("timestamp", "2nd_vacuum_timestamp"),
+            how="outer", on=["catalog", "database", "tableName"]
+        )
+        # summary of table metrics
+        table_metrics_1 = (
+            operation_order.filter((F.col("operation") == "OPTIMIZE") & (F.col("operation_order") == 1))
+            .select([
+                F.col("catalog"),
+                F.col("database"),
+                F.col("tableName"),
+                F.col("min_file_size"),
+                F.col("p50_file_size"),
+                F.col("max_file_size"),
+                F.col("z_order_by"),
+            ])
+        )
+
+        # write to output
+        out = out.join(
+            table_metrics_1,
+            how="outer", on=["catalog", "database", "tableName"]
+        )
+
+        return out
+
+    def scan(
+        self,
+        table_info_list: Iterable[TableInfo],
+        housekeeping_table_name: str = "lorenzorubi.default.housekeeping_summary_v2",  # TODO remove
+        do_save_as_table: bool = True,
+    ) -> pd.DataFrame:
+        """
+        Scans a table_info / table_info_list to fetch Delta stats
+        - DESCRIBE DETAIL
+        - DESCRIBE HISTORY
+        """
+        dd_list = []
+        statements = []
+        errors = []
+
+        if not isinstance(table_info_list, Iterable):
+            table_info_list = [table_info_list]
+
+        for table_info in table_info_list:
```

Comment:
You are not using the `map` function, this is not an iterable any more.
Please simplify this part by removing the for loop.